runtime/common/src/lib.rs

```diff
@@ -0,0 +1,72 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod impls;
+pub use types::*;
+pub use constants::*;
+
+/// Common types of statemint and statemine.
+mod types {
+	use sp_runtime::traits::{Verify, IdentifyAccount, BlakeTwo256};
+
+	/// An index to a block.
+	pub type BlockNumber = u32;
+
+	/// Alias to 512-bit hash when used in the context of a transaction signature on the chain.
+	pub type Signature = sp_runtime::MultiSignature;
+
+	/// Some way of identifying an account on the chain. We intentionally make it equivalent
+	/// to the public key of our transaction signing scheme.
+	pub type AccountId = <<Signature as Verify>::Signer as IdentifyAccount>::AccountId;
+
+	/// The type for looking up accounts. We don't expect more than 4 billion of them, but you
+	/// never know...
+	pub type AccountIndex = u32;
+
+	/// Balance of an account.
+	pub type Balance = u128;
+
+	/// Index of a transaction in the chain.
+	pub type Index = u32;
+
+	/// A hash of some data used by the chain.
+	pub type Hash = sp_core::H256;
+
+	/// Block header type as expected by this runtime.
+	pub type Header = sp_runtime::generic::Header<BlockNumber, BlakeTwo256>;
+
+	/// Digest item type.
+	pub type DigestItem = sp_runtime::generic::DigestItem<Hash>;
+	
+	// Aura consensus authority.
+	pub type AuraId = sp_consensus_aura::sr25519::AuthorityId;
+}
+
+/// Common constants of statemint and statemine
+mod constants {
+	use super::types::BlockNumber;
+	use sp_runtime::Perbill;
+	use frame_support::weights::{Weight, constants::WEIGHT_PER_SECOND};
+	/// This determines the average expected block time that we are targeting. Blocks will be
+	/// produced at a minimum duration defined by `SLOT_DURATION`. `SLOT_DURATION` is picked up by
+	/// `pallet_timestamp` which is in turn picked up by `pallet_aura` to implement `fn
+	/// slot_duration()`.
+	///
+	/// Change this to adjust the block time.
+	pub const MILLISECS_PER_BLOCK: u64 = 12000;
```

Comment:
Is 12 sec the new standard now for collator blocks?

Reply:
I just followed statemint's configuration, but how much time it will spend still very unstable, even minutes. Now it's 6_000.